### PR TITLE
Redis connection pool

### PR DESCRIPTION
**Pull request by mariusandra**
## Changes

- With this change each Redis query needs to finish before a new one is allowed (per thread ofc). This might slow down plugins by a bit, but should greatly increase stability of the pipeline. 
- If this works fine, we could add a few more workers to the pool to increase throughput. This will increase the used redis connections, getting us in trouble with Heroku again...

## Checklist

-   [ ] Updated Settings section in README.md, if settings are affected
-   [ ] Jest tests

### CODE REVIEW

File: src/db.ts [hunk 1]
```diff
@@ -36,17 +38,28 @@ export class DB {
...
 
     constructor(postgres: Pool, redis: Redis.Redis, kafkaProducer?: Producer, clickhouse?: ClickHouse) {
         this.postgres = postgres
-        this.redis = redis
         this.kafkaProducer = kafkaProducer
         this.clickhouse = clickhouse
+        this.redisPool = createPool<Redis.Redis>(
+            {
+                create: () => Promise.resolve(redis),
+                destroy: () => Promise.resolve(undefined),
```
Comment (Twixes) [inline src/db.ts:55]: Should this not stop/close/disconnect the Redis client?

File: src/db.ts [hunk 2]
```diff
@@ -36,17 +38,28 @@ export class DB {
...
+            },
+            {
+                min: 1,
+                max: 1,
```
Comment (Twixes) [inline src/db.ts:59]: With max 1 this doesn't really test pooling, how about 2?

File: src/db.ts (hunk 1 above)
Comment (mariusandra) [reply src/db.ts:55]: Not if using just one client and the pool is effectively a glorified lock.

File: src/db.ts (hunk 2 above)
Comment (mariusandra) [reply src/db.ts:59]: Not testing pooling, testing locking ;)

**Comment by mariusandra** [comment]
Let's actually go for a real pool...

**Comment by mariusandra** [comment]
Something's broken with celery now :/. Will investigate/fix later today.

**Comment by mariusandra** [comment]
Ready for a look! I had to change the tests around quite a bit, such as removing `ioredis-mock`, etc...

**Review by Twixes — APPROVED** [review]
Looks good!

**Pull request merged**